test/sequential/test-regress-GH-9765.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that if an Immediate callback clears subsequent
+// immediates we don't get stuck in an infinite loop.
+//
+// Ref: https://github.com/nodejs/node/issues/9756
```

Comment:
This probably isn't needed, the test runner has a timeout mechanism of it's own.

If you think it is, the other file should be in `/fixtures/`.

Either way, this file should probably be in `/parallel/`.